packages/node/src/stellar/api.stellar.ts

```diff
@@ -72,61 +85,241 @@ export class StellarApi implements ApiWrapper<StellarBlockWrapper> {
+
+    effects.forEach((effect) => {
+      const wrappedEffect: StellarEffect = {
+        ...effect,
+        ledger: null,
+        transaction: null,
+        operation: null,
+      };
+
+      wrappedEffects.push(wrappedEffect);
+    });
+
+    return wrappedEffects;
+  }
+
+  private async fetchOperationsForTransaction(
+    transactionId: string,
+    applicationOrder: number,
+    sequence: number,
+  ): Promise<StellarOperation[]> {
+    const operations = (
+      await this.api.operations().forTransaction(transactionId).call()
+    ).records;
+
+    return Promise.all(
+      operations.map(async (op) => {
+        //There will be only one operation for soroban transaction
+        let events: SorobanEvent[] = [];
+        try {
+          events =
+            op.type.toString() === 'invoke_host_function'
+              ? (await this.getAndWrapEvents(sequence)).filter(
+                  (event) =>
+                    this.getTransactionApplicationOrder(event.id) ===
+                    applicationOrder,
+                )
+              : ([] as SorobanEvent[]);
+        } catch (e) {
+          if (e.message === 'start is before oldest ledger') {
+            throw new Error(`The requested events for ledger number ${sequence} is not available on the current soroban node. 
+            This is because you're trying to access a ledger that is older than the oldest ledger stored in this node. 
+            To resolve this issue, you can either:
+            1. Increase the start ledger to a more recent one, or
+            2. Connect to a different node that might have a longer history of ledgers.`);
+          }
+
+          throw e;
+        }
+
+        const wrappedOp: StellarOperation = {
+          ...op,
+          ledger: null,
+          transaction: null,
+          effects: [] as StellarEffect[],
+          events: events,
+        };
+
+        const effects = (await this.fetchEffectsForOperation(op.id)).map(
+          (effect) => {
+            effect.operation = JSON.parse(JSON.stringify(wrappedOp));
+            return effect;
           },
-        },
-        topic: event.topic.map((topic) =>
-          StellarBlockWrapped.decodeScVals(xdr.ScVal.fromXDR(topic, 'base64')),
-        ),
-      }));
+        );
 
-      const ret = new StellarBlockWrapped(events, {
-        ledger: blockNumber,
-        hash: blockNumber.toString(),
-      } as StellarBlock);
+        wrappedOp.effects.push(...effects);
 
-      this.eventEmitter.emit('fetchBlock');
-      return ret;
-    } catch (e) {
-      throw this.handleError(e, blockNumber);
-    }
+        return wrappedOp;
+      }),
+    );
   }
 
-  async fetchBlocks(bufferBlocks: number[]): Promise<StellarBlockWrapper[]> {
+  private async fetchTransactionsForLedger(
+    sequence: number,
+  ): Promise<StellarTransaction[]> {
+    const transactions = (
+      await this.api.transactions().forLedger(sequence).call()
+    ).records;
```

Comment:
Rather than having parenthesis around await its much more readable to use a destructor. Theres a couple of places this can be improved
```suggestion
    const { records: transactions } = await this.api.transactions().forLedger(sequence).call();
```